pandas/_libs/index.pyx

```diff
@@ -491,6 +494,76 @@ cdef class TimedeltaEngine(DatetimeEngine):
     cdef _get_box_dtype(self):
         return 'm8[ns]'
 
+
+cdef class PeriodEngine(Int64Engine):
+
+    cdef _get_index_values(self):
+        return self.vgetter()
+
+    cpdef _call_map_locations(self, values):
+        super(PeriodEngine, self)._call_map_locations(values.view('i8'))
+
+    def _call_monotonic(self, values):
+        return super(PeriodEngine, self)._call_monotonic(values.view('i8'))
+
+    cdef _maybe_get_bool_indexer(self, object val):
```

Comment:
is this duplicating some of the super code?

Reply:
Almost same, but where getting the `values` is not same.
https://github.com/Licht-T/pandas/blob/master/pandas/_libs/index_class_helper.pxi.in#L69
https://github.com/pandas-dev/pandas/pull/17755/files#diff-495f440aad68721ad5ca1f35087fc450R520

Reply:
@jreback Almost same, but where getting the `values` is not same.
https://github.com/Licht-T/pandas/blob/master/pandas/_libs/index_class_helper.pxi.in#L69
https://github.com/pandas-dev/pandas/pull/17755/files#diff-495f440aad68721ad5ca1f35087fc450R520

Reply:
so provide a helper function to avoid repeating all of this code again.